### PR TITLE
Refactor voice module to the new architecture

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description
Refactors voice module to the new architecture

### Changelog
```
<changelog>Refactored voice module to the new architecture.</changelog>
```

### CODE REVIEW

**Comment by Guardiola31337** [comment]
Fixed https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/10625/workflows/b5c5b0cf-1f5c-4337-bf12-3a81b5c6621e/jobs/41513 in https://github.com/mapbox/mapbox-navigation-android/pull/4086/commits/dbbf77acc7acc269dfc49330dd790b029c1c99cc

cc @LukasPaczos @cafesilencio @abhishek1508

File: libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/model/SpeechAnnouncement.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ import java.io.File
  * @property ssmlAnnouncement SSML announcement text retrieved from [VoiceInstructions].
  * @property file synthesized speech mp3.
  */
-data class Announcement(
+data class SpeechAnnouncement(
```
Comment (Guardiola31337) [inline libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/model/SpeechAnnouncement.kt:10]: Should this be a builder? 🤔

File: libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/model/SpeechVolume.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.mapbox.navigation.ui.voice.model
+
+import androidx.annotation.FloatRange
+
+/**
+ * The state is returned if we change the speech volume.
+ * @param level
+ */
+class SpeechVolume(
```
Comment (Guardiola31337) [inline libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/model/SpeechVolume.kt:9]: Should this be a builder? 🤔 

We can also explore @cafesilencio's idea 

```kotlin
interface SomeValue {
    fun getValue(): String?
}
interface SomeValueError: SomeValue {
    fun getThrowable(): Throwable
}
interface SomeVolumeValue: SomeValue {
    fun getVolumeValue: SomeType
}
```

then in the future if there is something added

```kotlin
interface someValueV2: SomeValue {
    fun getAnotherValue(): Whatever
}
```

backwards compatible while also extensible.

**Review by Guardiola31337 — COMMENTED** [review]
Dropped some comments to discuss as a team @mapbox/navigation-android

**Review by LukasPaczos — APPROVED** [review]
LGTM

**Pull request merged**